QSV AV1 10-bit Errors

### Problem Description

Compressing low resolution(SD 480p) video files results in corrupt output. VLC trys to play it still but the video chanel is distorted nonsense. 

### Activity Log, Crash Log or any other details

```text
HandBrake Nightly 20221005173115-d026bcf17-master (2022100701)
OS: Microsoft Windows NT 10.0.19044.0
CPU: AMD Ryzen 9 5900X 12-Core Processor            
Ram: 32693 MB, 
GPU Information:
  NVIDIA GeForce GTX 1070 - 31.0.15.1740
  Intel(R) Arc(TM) A380 Graphics - 31.0.101.3430
Screen: 1920x1080
Temp Dir: C:\Users\skech\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrakeTeam\Nightly\
Data Dir: C:\Users\skech\AppData\Roaming\HandBrake\Nightly

-------------------------------------------

 # Starting Encode ...

[19:30:53] base preset: QSV AV1 Generic stereo (Modified)
[19:30:53] Remote Process started with Process ID: 15908 using port: 8037. Max Allowed Instances: 1
[19:30:53] Worker: Starting HandBrake Engine ...
[19:30:53] Worker: Starting Web Server on port 8037 ...
[19:30:53] Worker: Disconnected worker monitoring enabled!
[19:30:53] Compile-time hardening features are enabled
[19:30:54] vcn: not available on this system
[19:30:54] nvenc: version 11.1 is available
[19:30:54] nvdec: is available
[19:30:54] hb_init: starting libhb thread
[19:30:54] Starting work at: Fri Oct 07 19:30:54 2022
[19:30:54] 1 job(s) to process
[19:30:54] json job:
{
  "Audio": {
    "AudioList": [
      {
        "DRC": 0,
        "Encoder": "vorbis",
        "Gain": 0,
        "Mixdown": 7,
        "NormalizeMixLevel": false,
        "Quality": 10,
        "Samplerate": 0,
        "Name": "Surround 5.1",
        "Track": 0,
        "DitherMethod": 0
      },
      {
        "DRC": 0,
        "Encoder": "vorbis",
        "Gain": 0,
        "Mixdown": 7,
        "NormalizeMixLevel": false,
        "Quality": 10,
        "Samplerate": 0,
        "Name": "Surround 5.1",
        "Track": 1,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:opus"
    ]
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      },
      {
        "Name": "Chapter 2"
      },
      {
        "Name": "Chapter 3"
      },
      {
        "Name": "Chapter 4"
      },
      {
        "Name": "Chapter 5"
      },
      {
        "Name": "Chapter 6"
      },
      {
        "Name": "Chapter 7"
      },
      {
        "Name": "Chapter 8"
      },
      {
        "Name": "Chapter 9"
      },
      {
        "Name": "Chapter 10"
      },
      {
        "Name": "Chapter 11"
      },
      {
        "Name": "Chapter 12"
      },
      {
        "Name": "Chapter 13"
      },
      {
        "Name": "Chapter 14"
      },
      {
        "Name": "Chapter 15"
      },
      {
        "Name": "Chapter 16"
      },
      {
        "Name": "Chapter 17"
      },
      {
        "Name": "Chapter 18"
      },
      {
        "Name": "Chapter 19"
      },
      {
        "Name": "Chapter 20"
      },
      {
        "Name": "Chapter 21"
      },
      {
        "Name": "Chapter 22"
      },
      {
        "Name": "Chapter 23"
      },
      {
        "Name": "Chapter 24"
      },
      {
        "Name": "Chapter 25"
      },
      {
        "Name": "Chapter 26"
      },
      {
        "Name": "Chapter 27"
      },
      {
        "Name": "Chapter 28"
      },
      {
        "Name": "Chapter 29"
      },
      {
        "Name": "Chapter 30"
      },
      {
        "Name": "Chapter 31"
      },
      {
        "Name": "Chapter 32"
      },
      {
        "Name": "Chapter 33"
      },
      {
        "Name": "Chapter 34"
      },
      {
        "Name": "Chapter 35"
      },
      {
        "Name": "Chapter 36"
      },
      {
        "Name": "Chapter 37"
      },
      {
        "Name": "Chapter 38"
      },
      {
        "Name": "Chapter 39"
      },
      {
        "Name": "Chapter 40"
      },
      {
        "Name": "Chapter 41"
      },
      {
        "Name": "Chapter 42"
      },
      {
        "Name": "Chapter 43"
      },
      {
        "Name": "Chapter 44"
      },
      {
        "Name": "Chapter 45"
      },
      {
        "Name": "Chapter 46"
      },
      {
        "Name": "Chapter 47"
      },
      {
        "Name": "Chapter 48"
      },
      {
        "Name": "Chapter 49"
      },
      {
        "Name": "Chapter 50"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": false,
    "File": "C:\\Users\\skech\\Videos\\Fratello test-10bit.mkv",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mkv"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 5,
        "Settings": {
          "mode": "11"
        }
      },
      {
        "ID": 3,
        "Settings": {
          "block-height": "16",
          "block-thresh": "40",
          "block-width": "16",
          "filter-mode": "2",
          "mode": "3",
          "motion-thresh": "1",
          "spatial-metric": "2",
          "spatial-thresh": "1"
        }
      },
      {
        "ID": 14,
        "Settings": {
          "crop-bottom": "0",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "0",
          "height": "480",
          "width": "720"
        }
      },
      {
        "ID": 7,
        "Settings": {
          "mode": "0"
        }
      }
    ]
  },
  "PAR": {
    "Num": 32,
    "Den": 27
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 7
    },
    "Title": 1,
    "Path": "\\\\NAS\\Video\\Raw Rips\B1_t00.mkv"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": [
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 1,
        "Offset": 0,
        "Track": 0
      },
      {
        "Burn": false,
        "Default": false,
        "Forced": false,
        "ID": 2,
        "Offset": 0,
        "Track": 1
      }
    ]
  },
  "Video": {
    "Encoder": "qsv_av1_10bit",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "lowpower=1",
    "Preset": "quality",
    "Profile": "auto",
    "Quality": 18,
    "QSV": {
      "Decode": true
    },
    "HardwareDecode": 4
  }
}
[19:30:54] CPU: AMD Ryzen 9 5900X 12-Core Processor
[19:30:54]  - logical processor count: 24
[19:30:54] Intel Quick Sync Video support: yes, gpu list: 0
[19:30:54] Intel Quick Sync Video discrete adapter with index 0
[19:30:54] Impl mfx-gen library path: C:\Windows\System32\DriverStore\FileRepository\iigd_dch_d.inf_amd64_bbe3135af798f492\libmfx64-gen.dll
[19:30:54]  - Intel Media SDK hardware: API 2.7 (minimum: 1.3)
[19:30:54]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes) av1 (8bit: yes, 10bit: yes)
[19:30:54]  - H.264 encoder: yes
[19:30:54]     - preferred implementation: hardware (1) via D3D11
[19:30:54]     - capabilities (hardware):  lowpower breftype icq vsinfo chromalocinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[19:30:54]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[19:30:54]     - preferred implementation: hardware (1) via D3D11
[19:30:54]     - capabilities (hardware):  lowpower bpyramid icq vsinfo masteringinfo cllinfo opt1 hyperencode
[19:30:54]  - AV1 encoder: yes (8bit: yes, 10bit: yes)
[19:30:54]     - preferred implementation: hardware (1) via D3D11
[19:30:54]     - capabilities (hardware):  lowpower bpyramid icq vsinfo opt1 av1bitstream
[19:30:54] hb_scan: path=\\NAS\Video\Raw Rips\B1_t00.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image \\NAS\Video\Raw Rips\B1_t00.mkv
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
[19:30:54] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
libdvdrealibdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
d: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[19:30:54] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from '\\NAS\Video\Raw Rips\B1_t00.mkv':
  Metadata:
    encoder         : libmakemkv v1.17.1 (1.3.10/1.5.2) x86_64-pc-linux-gnu
    creation_time   : 2022-07-31T20:30:41.000000Z
  Duration: 02:42:36.81, start: 0.000000, bitrate: 6455 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 111.044267
      Metadata:
        title           : Chapter 01
    Chapter #0:1: start 111.044267, end 168.601767
      Metadata:
        title           : Chapter 02
    Chapter #0:2: start 168.601767, end 366.899867
      Metadata:
        title           : Chapter 03
    Chapter #0:3: start 366.899867, end 680.012667
      Metadata:
        title           : Chapter 04
    Chapter #0:4: start 680.012667, end 984.316667
      Metadata:
        title           : Chapter 05
    Chapter #0:5: start 984.316667, end 1311.076433
      Metadata:
        title           : Chapter 06
    Chapter #0:6: start 1311.076433, end 1392.424367
      Metadata:
        title           : Chapter 07
    Chapter #0:7: start 1392.424367, end 1503.468633
      Metadata:
        title           : Chapter 08
    Chapter #0:8: start 1503.468633, end 1535.033500
      Metadata:
        title           : Chapter 09
    Chapter #0:9: start 1535.033500, end 1715.714000
      Metadata:
        title           : Chapter 10
    Chapter #0:10: start 1715.714000, end 1988.453133
      Metadata:
        title           : Chapter 11
    Chapter #0:11: start 1988.453133, end 2363.427733
      Metadata:
        title           : Chapter 12
    Chapter #0:12: start 2363.427733, end 2703.434067
      Metadata:
        title           : Chapter 13
    Chapter #0:13: start 2703.434067, end 2784.815367
      Metadata:
        title           : Chapter 14
    Chapter #0:14: start 2784.815367, end 2895.893000
      Metadata:
        title           : Chapter 15
    Chapter #0:15: start 2895.893000, end 3006.837167
      Metadata:
        title           : Chapter 16
    Chapter #0:16: start 3006.837167, end 3252.749500
      Metadata:
        title           : Chapter 17
    Chapter #0:17: start 3252.749500, end 3346.843500
      Metadata:
        title           : Chapter 18
    Chapter #0:18: start 3346.843500, end 3665.795467
      Metadata:
        title           : Chapter 19
    Chapter #0:19: start 3665.795467, end 4095.658233
      Metadata:
        title           : Chapter 20
    Chapter #0:20: start 4095.658233, end 4177.206367
      Metadata:
        title           : Chapter 21
    Chapter #0:21: start 4177.206367, end 4288.250633
      Metadata:
        title           : Chapter 22
    Chapter #0:22: start 4288.250633, end 4397.226167
      Metadata:
        title           : Chapter 23
    Chapter #0:23: start 4397.226167, end 4521.149967
      Metadata:
        title           : Chapter 24
    Chapter #0:24: start 4521.149967, end 4884.212667
      Metadata:
        title           : Chapter 25
    Chapter #0:25: start 4884.212667, end 5212.006800
      Metadata:
        title           : Chapter 26
    Chapter #0:26: start 5212.006800, end 5488.216067
      Metadata:
        title           : Chapter 27
    Chapter #0:27: start 5488.216067, end 5569.597367
      Metadata:
        title           : Chapter 28
    Chapter #0:28: start 5569.597367, end 5680.675000
      Metadata:
        title           : Chapter 29
    Chapter #0:29: start 5680.675000, end 5795.489700
      Metadata:
        title           : Chapter 30
    Chapter #0:30: start 5795.489700, end 5965.759800
      Metadata:
        title           : Chapter 31
    Chapter #0:31: start 5965.759800, end 6222.683133
      Metadata:
        title           : Chapter 32
    Chapter #0:32: start 6222.683133, end 6608.835567
      Metadata:
        title           : Chapter 33
    Chapter #0:33: start 6608.835567, end 6880.673800
      Metadata:
        title           : Chapter 34
    Chapter #0:34: start 6880.673800, end 6961.988367
      Metadata:
        title           : Chapter 35
    Chapter #0:35: start 6961.988367, end 7073.032633
      Metadata:
        title           : Chapter 36
    Chapter #0:36: start 7073.032633, end 7086.479400
      Metadata:
        title           : Chapter 37
    Chapter #0:37: start 7086.479400, end 7368.761400
      Metadata:
        title           : Chapter 38
    Chapter #0:38: start 7368.761400, end 7655.014033
      Metadata:
        title           : Chapter 39
    Chapter #0:39: start 7655.014033, end 7952.411133
      Metadata:
        title           : Chapter 40
    Chapter #0:40: start 7952.411133, end 8273.064800
      Metadata:
        title           : Chapter 41
    Chapter #0:41: start 8273.064800, end 8354.379367
      Metadata:
        title           : Chapter 42
    Chapter #0:42: start 8354.379367, end 8465.457000
      Metadata:
        title           : Chapter 43
    Chapter #0:43: start 8465.457000, end 8536.027500
      Metadata:
        title           : Chapter 44
    Chapter #0:44: start 8536.027500, end 8780.772000
      Metadata:
        title           : Chapter 45
    Chapter #0:45: start 8780.772000, end 9094.485400
      Metadata:
        title           : Chapter 46
    Chapter #0:46: start 9094.485400, end 9372.329633
      Metadata:
        title           : Chapter 47
    Chapter #0:47: start 9372.329633, end 9665.455800
      Metadata:
        title           : Chapter 48
    Chapter #0:48: start 9665.455800, end 9746.770367
      Metadata:
        title           : Chapter 49
    Chapter #0:49: start 9746.770367, end 9756.813733
      Metadata:
        title           : Chapter 50
  Stream #0:0(eng): Video: mpeg2video (Main), yuv420p(tv, progressive), 720x480 [SAR 32:27 DAR 16:9], SAR 853:720 DAR 853:480, 29.97 fps, 29.97 tbr, 1k tbn
    Metadata:
      BPS-eng         : 5551010
      DURATION-eng    : 02:42:36.813733333
      NUMBER_OF_FRAMES-eng: 233930
      NUMBER_OF_BYTES-eng: 6770021536
      SOURCE_ID-eng   : 0100E0
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.17.1 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2022-07-31 20:30:41
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
    Side data:
      cpb: bitrate max/min/avg: 9000000/0/0 buffer size: 1835008 vbv_delay: N/A
  Stream #0:1(eng): Audio: ac3, 48000 Hz, 5.1(side), fltp, 448 kb/s (default)
    Metadata:
      title           : Surround 5.1
      BPS-eng         : 447988
      DURATION-eng    : 02:42:36.754366666
      NUMBER_OF_FRAMES-eng: 304891
      NUMBER_OF_BYTES-eng: 546364672
      SOURCE_ID-eng   : 0180BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.17.1 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2022-07-31 20:30:41
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:2(jpn): Audio: ac3, 48000 Hz, 5.1(side), fltp, 448 kb/s
    Metadata:
      title           : Surround 5.1
      BPS-eng         : 447737
      DURATION-eng    : 02:42:36.754366666
      NUMBER_OF_FRAMES-eng: 304720
      NUMBER_OF_BYTES-eng: 546058240
      SOURCE_ID-eng   : 0181BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.17.1 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2022-07-31 20:30:41
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:3(eng): Subtitle: dvd_subtitle, 720x480 (default)
    Metadata:
      BPS-eng         : 20
      DURATION-eng    : 02:39:44.240688888
      NUMBER_OF_FRAMES-eng: 16
      NUMBER_OF_BYTES-eng: 23992
      SOURCE_ID-eng   : 0120BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.17.1 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2022-07-31 20:30:41
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
  Stream #0:4(eng): Subtitle: dvd_subtitle, 720x480
    Metadata:
      BPS-eng         : 3249
      DURATION-eng    : 02:40:10.600355555
      NUMBER_OF_FRAMES-eng: 1486
      NUMBER_OF_BYTES-eng: 3903532
      SOURCE_ID-eng   : 0121BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.17.1 linux(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2022-07-31 20:30:41
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
[19:30:54] scan: decoding previews for title 1
[19:30:54] scan: audio 0x1: ac3, rate=48000Hz, bitrate=448000 English (AC3) (5.1 ch) (448 kbps)
[19:30:54] scan: audio 0x2: ac3, rate=48000Hz, bitrate=448000 ??? (AC3) (5.1 ch) (448 kbps)
[19:30:54] WARNING: Video PAR 32:27 != container PAR 853:720
[19:30:54] scan: 10 previews, 720x480, 23.976 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 32:27, color profile: 6-1-6, chroma location: left
[19:30:54] scan: supported video decoders: avcodec nvdec
[19:30:55] libhb: scan thread found 1 valid title(s)
[19:30:55] qsv: discrete qsv adapter with index 0 has been selected
[19:30:55] Starting Task: Encoding Pass
[19:30:55] Skipping vfr filter
[19:30:55] Skipping crop/scale filter
[19:30:55] work: track 1, dithering not supported by codec
[19:30:55] work: track 2, dithering not supported by codec
[19:30:55] job configuration:
[19:30:55]  * source
[19:30:55]    + \\NAS\Video\Raw Rips\B1_t00.mkv
[19:30:55]    + title 1, chapter(s) 1 to 7
[19:30:55]    + container: matroska,webm
[19:30:55]    + data rate: 6455 kbps
[19:30:55]  * destination
[19:30:55]    + C:\Users\skech\Videos\Fratello test-10bit.mkv
[19:30:55]    + container: Matroska (libavformat)
[19:30:55]      + chapter markers
[19:30:55]  * video track
[19:30:55]    + decoder: mpeg2video 8-bit (yuv420p)
[19:30:55]    + filters
[19:30:55]      + Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
[19:30:55]      + Deinterlace (mode=43)
[19:30:55]      + Format (format=p010le)
[19:30:55]    + Output geometry
[19:30:55]      + storage dimensions: 720 x 480
[19:30:55]      + pixel aspect ratio: 32 : 27
[19:30:55]      + display dimensions: 853 x 480
[19:30:55]    + encoder: AV1 10-bit (Intel Media SDK)
[19:30:55]      + preset:  quality
[19:30:55]      + options: lowpower=1
[19:30:55]      + profile: auto
[19:30:55]      + level:   auto
[19:30:55]      + quality: 18.00 (ICQ)
[19:30:55]      + color profile: 6-1-6
[19:30:55]      + chroma location: left
[19:30:55]  * subtitle track 1, English [VOBSUB] (track 0, id 0x3, Picture) -> Passthrough
[19:30:55]  * subtitle track 2, English [VOBSUB] (track 1, id 0x4, Picture) -> Passthrough
[19:30:55]  * audio track 1
[19:30:55]    + name: Surround 5.1
[19:30:55]    + decoder: English (AC3) (5.1 ch) (448 kbps) (track 1, id 0x1)
[19:30:55]      + bitrate: 448 kbps, samplerate: 48000 Hz
[19:30:55]    + mixdown: 5.1 Channels
[19:30:55]    + encoder: Vorbis (libvorbis)
[19:30:55]      + quality: 10.00, samplerate: 48000 Hz
[19:30:55]  * audio track 2
[19:30:55]    + name: Surround 5.1
[19:30:55]    + decoder: ??? (AC3) (5.1 ch) (448 kbps) (track 2, id 0x2)
[19:30:55]      + bitrate: 448 kbps, samplerate: 48000 Hz
[19:30:55]    + mixdown: 5.1 Channels
[19:30:55]    + encoder: Vorbis (libvorbis)
[19:30:55]      + quality: 10.00, samplerate: 48000 Hz
[19:30:55] sync: expecting 33384 video frames
[19:30:55] encvorbis: opening libvorbis
[19:30:55] encvorbis: opening libvorbis
[19:30:55] Unknown Header.BufferId=1396846913
[19:30:55] encqsvInit: using encode-only (LowPower) path
[19:30:55] encqsvInit: AV1 Main profile @ level 3.0
[19:30:55] encqsvInit: TargetUsage 1 AsyncDepth 6
[19:30:55] encqsvInit: GopRefDist 4 GopPicSize 24 NumRefFrame 2
[19:30:55] encqsvInit: BFramesMax 3 BRefType pyramid
[19:30:55] encqsvInit: RateControlMethod ICQ ICQQuality 18
[19:30:55] sync: first pts video is 0
[19:30:55] sync: "Chapter 1" (1) at frame 1 time 0
[19:30:55] sync: first pts audio 0x1 is 0
[19:30:55] sync: first pts audio 0x2 is 0
[19:30:55] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 2.7
[19:30:59] sync: "Chapter 2" (2) at frame 2665 time 9999990
[19:31:00] sync: first pts subtitle 0x4 is 12297330
[19:31:01] sync: first pts subtitle 0x3 is 14669640
[19:31:02] sync: "Chapter 3" (3) at frame 4043 time 15172657
[19:31:10] sync: "Chapter 4" (4) at frame 8798 time 33021738
[19:31:22] sync: "Chapter 5" (5) at frame 16305 time 61201140
[19:31:34] sync: "Chapter 6" (6) at frame 23601 time 88588500
[19:31:48] sync: "Chapter 7" (7) at frame 31437 time 118002885
[19:31:51] reader: end of chapter 7 (media 7) reached at media chapter 8
[19:31:51] reader: done. 1 scr changes
[19:31:51] sync: "Chapter 8" (8) at frame 33387 time 125325720
[19:31:51] work: average encoding speed for job is 597.639404 fps
[19:31:51] comb detect: heavy 141 | light 611 | uncombed 32635 | total 33387
[19:31:51] ac3-decoder done: 43516 frames, 0 decoder errors
[19:31:51] ac3-decoder done: 43495 frames, 0 decoder errors
[19:31:51] mpeg2video-decoder done: 33387 frames, 0 decoder errors
[19:31:51] sync: got 33387 frames, 33384 expected
[19:31:51] sync: framerate min 13.280 fps, max 29.970 fps, avg 23.976 fps
[19:31:51] mux: track 0, 33387 frames, 212004272 bytes, 1217.93 kbps, fifo 1024
[19:31:51] mux: track 1, 165265 frames, 188035446 bytes, 1080.23 kbps, fifo 8192
[19:31:51] mux: track 2, 155519 frames, 183955769 bytes, 1056.79 kbps, fifo 8192
[19:31:51] mux: track 3, 2 frames, 3092 bytes, 0.02 kbps, fifo 8
[19:31:51] mux: track 4, 184 frames, 461308 bytes, 2.65 kbps, fifo 16
[19:31:51] Finished work at: Fri Oct 07 19:31:51 2022
[19:31:51] libhb: work result = 0

 # Job Completed!
```


### What Operating System are you running?

Windows 10 Pro

### What version of HandBrake are you running?

Nightly 20221005173115-d026bcf17-master (2022100701)

### Where did you download HandBrake from?

github

HandBrake Nightly 20221005173115-d026bcf17-master (2022100701)
OS: Microsoft Windows NT 10.0.22000.0
CPU: 12th Gen Intel(R) Core(TM) i7-12700H
Ram: 32492 MB, 
GPU Information:
  Intel(R) Arc(TM) A370M Graphics - 31.0.101.3430
  Intel(R) Iris(R) Xe Graphics - 31.0.101.3430
Screen: 2048x1280
Temp Dir: C:\Users\Scott\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake Nightly\
Data Dir: C:\Users\Scott\AppData\Roaming\HandBrake\Nightly

-------------------------------------------

 # Starting Encode ...

[17:30:13] base preset: QSV Test (Modified)
[17:30:13] Remote Process started with Process ID: 20384 using port: 8038. Max Allowed Instances: 1
[17:30:13] Worker: Starting HandBrake Engine ...
[17:30:13] Worker: Starting Web Server on port 8038 ...
[17:30:14] Worker: Disconnected worker monitoring enabled!
[17:30:14] Compile-time hardening features are enabled
[17:30:16] vcn: not available on this system
Cannot load nvEncodeAPI64.dll
[17:30:16] hb_init: starting libhb thread
[17:30:16] Starting work at: Sun Oct 09 17:30:16 2022
[17:30:16] 1 job(s) to process
[17:30:16] json job:
{
  "Audio": {
    "AudioList": [],
    "CopyMask": [
      "copy:aac"
    ]
  },
  "Destination": {
    "ChapterList": [],
    "ChapterMarkers": false,
    "AlignAVStart": false,
    "File": "C:\\Users\\Scott\\Desktop\\Output\\Kj(1).mkv",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mkv"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 14,
        "Settings": {
          "crop-bottom": "74",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "74",
          "height": "428",
          "width": "720"
        }
      },
      {
        "ID": 7,
        "Settings": {
          "mode": "2",
          "rate": "27000000/900000"
        }
      }
    ]
  },
  "PAR": {
    "Num": 64,
    "Den": 45
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "C:\\Users\\Scott\\Videos\\KJ.ISO"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_av1_10bit",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "hyperencode=adaptive:lowpower=1",
    "Preset": "balanced",
    "Profile": "auto",
    "Quality": 22,
    "QSV": {
      "Decode": true
    },
    "HardwareDecode": 4
  }
}
[17:30:16] CPU: 12th Gen Intel(R) Core(TM) i7-12700H
[17:30:16]  - Intel microarchitecture Alder Lake performance hybrid architecture
[17:30:16]  - logical processor count: 20
[17:30:16] Intel Quick Sync Video support: yes, gpu list: 0, 1
[17:30:16] Intel Quick Sync Video integrated adapter with index 0
[17:30:16] Impl mfx-gen library path: C:\Windows\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_d7ee3f80fe1ead36\libmfx64-gen.dll
[17:30:16]  - Intel Media SDK hardware: API 2.7 (minimum: 1.3)
[17:30:16]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes) av1 (8bit: yes, 10bit: yes)
[17:30:16]  - H.264 encoder: yes
[17:30:16]     - preferred implementation: hardware (1) via D3D11
[17:30:16]     - capabilities (hardware):  lowpower breftype icq vsinfo chromalocinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[17:30:16]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[17:30:16]     - preferred implementation: hardware (1) via D3D11
[17:30:16]     - capabilities (hardware):  lowpower bpyramid icq vsinfo masteringinfo cllinfo opt1 hyperencode
[17:30:16]  - AV1 encoder: no
[17:30:16] Intel Quick Sync Video discrete adapter with index 1
[17:30:16] Impl mfx-gen library path: C:\Windows\System32\DriverStore\FileRepository\iigd_dch_d.inf_amd64_bbe3135af798f492\libmfx64-gen.dll
[17:30:16]  - Intel Media SDK hardware: API 2.7 (minimum: 1.3)
[17:30:16]  - Decode support:  h264 hevc (8bit: yes, 10bit: yes) av1 (8bit: yes, 10bit: yes)
[17:30:16]  - H.264 encoder: yes
[17:30:16]     - preferred implementation: hardware (2) via D3D11
[17:30:16]     - capabilities (hardware):  lowpower breftype icq vsinfo chromalocinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[17:30:16]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[17:30:16]     - preferred implementation: hardware (2) via D3D11
[17:30:16]     - capabilities (hardware):  lowpower bpyramid icq vsinfo masteringinfo cllinfo opt1 hyperencode
[17:30:16]  - AV1 encoder: yes (8bit: yes, 10bit: yes)
[17:30:16]     - preferred implementation: hardware (2) via D3D11
[17:30:16]     - capabilities (hardware):  lowpower bpyramid icq vsinfo opt1 av1bitstream
[17:30:16] hb_scan: path=C:\Users\Scott\Videos\KJ.ISO, title_index=1
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
[17:30:16] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
libdvdnav: DVD Title: KANGAROO_JACK
libdvdnav: DVD Serial Number: 2F058B0F
libdvdnav: DVD Title (Alternative): KANGAROO_JACK
libdvdnav: DVD disk reports itself with Region mask 0x00000000. Regions: 01 02 03 04 05 06 07 08
libdvdread: Encrypted DVD support unavailable.
[17:30:16] scan: DVD has 10 title(s)
[17:30:16] scan: scanning title 1
[17:30:16] scan: duration is 01:25:25 (5125120 ms)
[17:30:16] pgc_id: 1, pgn: 1: pgc: 00000289a487eed0
[17:30:16] scan: checking audio 1
[17:30:16] scan: id=0x80bd, lang=English (AC3), 3cc=eng ext=0
[17:30:16] scan: checking audio 2
[17:30:16] scan: id=0x81bd, lang=Deutsch (AC3), 3cc=deu ext=0
[17:30:16] scan: checking audio 3
[17:30:16] scan: id=0x82bd, lang=espaol (AC3), 3cc=spa ext=0
[17:30:16] scan: checking audio 4
[17:30:16] scan: id=0x83bd, lang=English (AC3), 3cc=eng ext=3
[17:30:16] scan: checking audio 5
[17:30:16] scan: audio channel is not active
[17:30:16] scan: checking subtitle 1
[17:30:16] scan: id=0x20bd, lang=English (Wide Screen) [VOBSUB], 3cc=eng ext=0
[17:30:16] scan: checking subtitle 2
[17:30:16] scan: id=0x21bd, lang=espaol (Wide Screen) [VOBSUB], 3cc=spa ext=0
[17:30:16] scan: checking subtitle 3
[17:30:16] scan: id=0x22bd, lang=Deutsch (Wide Screen) [VOBSUB], 3cc=deu ext=0
libdvdnav: DVD disk reports itself with Region mask 0x00000000. Regions: 01 02 03 04 05 06 07 08
[17:30:16] scan: checking subtitle 4
[17:30:16] scan: id=0x23bd, lang=Francais (Wide Screen) [VOBSUB], 3cc=fra ext=0
[17:30:16] scan: checking subtitle 5
[17:30:16] scan: id=0x24bd, lang=italiano (Wide Screen) [VOBSUB], 3cc=ita ext=0
[17:30:16] scan: checking subtitle 6
[17:30:16] scan: id=0x25bd, lang=Portugues (Wide Screen) [VOBSUB], 3cc=por ext=0
[17:30:16] scan: checking subtitle 7
[17:30:16] scan: id=0x26bd, lang=dansk (Wide Screen) [VOBSUB], 3cc=dan ext=0
[17:30:16] scan: checking subtitle 8
[17:30:16] scan: id=0x27bd, lang=suomi (Wide Screen) [VOBSUB], 3cc=fin ext=0
[17:30:16] scan: checking subtitle 9
[17:30:16] scan: id=0x28bd, lang=­slenska (Wide Screen) [VOBSUB], 3cc=isl ext=0
[17:30:16] scan: checking subtitle 10
[17:30:16] scan: id=0x29bd, lang=norsk (Wide Screen) [VOBSUB], 3cc=nor ext=0
[17:30:16] scan: checking subtitle 11
[17:30:16] scan: id=0x2abd, lang=svenska (Wide Screen) [VOBSUB], 3cc=swe ext=0
[17:30:16] scan: checking subtitle 12
[17:30:16] scan: id=0x2bbd, lang=Œ??????? (Wide Screen) [VOBSUB], 3cc=ell ext=0
[17:30:16] scan: checking subtitle 13
[17:30:16] scan: id=0x2cbd, lang=English (Wide Screen) [VOBSUB], 3cc=eng ext=1
[17:30:16] scan: checking subtitle 14
[17:30:16] scan: id=0x2dbd, lang=Deutsch (Wide Screen) [VOBSUB], 3cc=deu ext=1
[17:30:16] scan: title 1 has 28 chapters
[17:30:16] scan: chap 1, 200920 ms
[17:30:16] scan: chap 2, 256120 ms
[17:30:16] scan: chap 3, 132160 ms
[17:30:16] scan: chap 4, 213640 ms
[17:30:16] scan: chap 5, 142680 ms
[17:30:16] scan: chap 6, 132040 ms
[17:30:16] scan: chap 7, 173280 ms
[17:30:16] scan: chap 8, 276160 ms
[17:30:16] scan: chap 9, 201840 ms
[17:30:16] scan: chap 10, 232880 ms
[17:30:16] scan: chap 11, 274760 ms
[17:30:16] scan: chap 12, 196400 ms
[17:30:16] scan: chap 13, 91560 ms
[17:30:16] scan: chap 14, 224200 ms
[17:30:16] scan: chap 15, 91400 ms
[17:30:16] scan: chap 16, 246600 ms
[17:30:16] scan: chap 17, 145200 ms
[17:30:16] scan: chap 18, 139760 ms
[17:30:16] scan: chap 19, 165160 ms
[17:30:16] scan: chap 20, 147000 ms
[17:30:16] scan: chap 21, 126440 ms
[17:30:16] scan: chap 22, 42840 ms
[17:30:16] scan: chap 23, 240800 ms
[17:30:16] scan: chap 24, 109920 ms
[17:30:16] scan: chap 25, 236440 ms
[17:30:16] scan: chap 26, 249480 ms
[17:30:16] scan: chap 27, 131960 ms
[17:30:16] scan: chap 28, 303480 ms
[17:30:16] scan: aspect = 16:9
[17:30:16] scan: decoding previews for title 1
[17:30:16] scan: title angle(s) 1
[17:30:16] scan: audio 0x83bd: ac3, rate=48000Hz, bitrate=192000 English (AC3) (Director's Commentary 1) (2.0 ch) (Dolby Surround) (192 kbps)
[17:30:16] scan: audio 0x82bd: ac3, rate=48000Hz, bitrate=384000 espaol (AC3) (5.1 ch) (384 kbps)
[17:30:16] scan: audio 0x81bd: ac3, rate=48000Hz, bitrate=384000 Deutsch (AC3) (5.1 ch) (384 kbps)
[17:30:16] scan: audio 0x80bd: ac3, rate=48000Hz, bitrate=384000 English (AC3) (5.1 ch) (384 kbps)
[17:30:16] scan: 10 previews, 720x576, 25.000 fps, autocrop = 74/74/0/0, aspect 16:9, PAR 64:45, color profile: 5-1-6, chroma location: left
[17:30:16] libhb: scan thread found 1 valid title(s)
[17:30:17] qsv: discrete qsv adapter with index 1 has been selected
[17:30:17] Starting Task: Encoding Pass
[17:30:17] job configuration:
[17:30:17]  * source
[17:30:17]    + C:\Users\Scott\Videos\KJ.ISO
[17:30:17]    + title 1, chapter(s) 1 to 1
[17:30:17]  * destination
[17:30:17]    + C:\Users\Scott\Desktop\Output\Kj(1).mkv
[17:30:17]    + container: Matroska (libavformat)
[17:30:17]  * video track
[17:30:17]    + decoder: mpeg2video 8-bit (yuv420p)
[17:30:17]      + bitrate 200 kbps
[17:30:17]    + filters
[17:30:17]      + Framerate Shaper (mode=2:rate=27000000/900000)
[17:30:17]        + frame rate: 25.000 fps -> peak rate limited to 30.000 fps
[17:30:17]      + Crop and Scale (width=720:height=428:crop-top=74:crop-bottom=74:crop-left=0:crop-right=0)
[17:30:17]        + source: 720 * 576, crop (74/74/0/0): 720 * 428, scale: 720 * 428
[17:30:17]      + Format (format=p010le)
[17:30:17]    + Output geometry
[17:30:17]      + storage dimensions: 720 x 428
[17:30:17]      + pixel aspect ratio: 64 : 45
[17:30:17]      + display dimensions: 1024 x 428
[17:30:17]    + encoder: AV1 10-bit (Intel Media SDK)
[17:30:17]      + preset:  balanced
[17:30:17]      + options: hyperencode=adaptive:lowpower=1
[17:30:17]      + profile: auto
[17:30:17]      + level:   auto
[17:30:17]      + quality: 22.00 (ICQ)
[17:30:17]      + color profile: 5-1-6
[17:30:17]      + chroma location: left
libdvdread: Encrypted DVD support unavailable.
libdvdnav: DVD Title: KANGAROO_JACK
libdvdnav: DVD Serial Number: 2F058B0F
libdvdnav: DVD Title (Alternative): KANGAROO_JACK
libdvdnav: DVD disk reports itself with Region mask 0x00000000. Regions: 01 02 03 04 05 06 07 08
libdvdread: Encrypted DVD support unavailable.
libdvdnav: DVD disk reports itself with Region mask 0x00000000. Regions: 01 02 03 04 05 06 07 08
[17:30:17] sync: expecting 5023 video frames
[17:30:17] encqsvInit: hb_qsv_param_parse: unsupported option hyperencode
[17:30:17] Unknown Header.BufferId=1396846913
[17:30:17] encqsvInit: using encode-only (LowPower) path
[17:30:17] encqsvInit: AV1 Main profile @ level 3.0
[17:30:17] encqsvInit: TargetUsage 4 AsyncDepth 6
[17:30:17] encqsvInit: GopRefDist 4 GopPicSize 28 NumRefFrame 3
[17:30:17] encqsvInit: BFramesMax 3 BRefType pyramid
[17:30:17] encqsvInit: RateControlMethod ICQ ICQQuality 22
[17:30:17] sync: first pts video is 0
[17:30:17] sync: "Chapter 1" (1) at frame 1 time 0
[17:30:17] qsv_enc_init: using 'hardware (2) via D3D11' implementation, API: 2.7
[17:30:28] reader: end of chapter 1 (media 1) reached at media chapter 2
[17:30:28] reader: done. 1 scr changes
[17:30:28] work: average encoding speed for job is 459.286896 fps
[17:30:28] vfr: 5021 frames output, 0 dropped and 0 duped for CFR/PFR
[17:30:28] vfr: lost time: 0 (0 frames)
[17:30:28] vfr: gained time: 0 (0 frames) (0 not accounted for)
[17:30:28] mpeg2video-decoder done: 5023 frames, 0 decoder errors
[17:30:28] sync: got 5021 frames, 5023 expected
[17:30:28] sync: framerate min 25.000 fps, max 25.000 fps, avg 25.000 fps
[17:30:28] mux: track 0, 5021 frames, 58037801 bytes, 2311.34 kbps, fifo 2048
[17:30:28] Finished work at: Sun Oct 09 17:30:28 2022
[17:30:28] libhb: work result = 0

 # Job Completed!

--- a/libhb/decavcodec.c
+++ b/libhb/decavcodec.c
@@ -1249,7 +1249,7 @@ int reinit_video_filters(hb_work_private_t * pv)
         {
             hb_dict_set(settings, "w", hb_value_int(orig_width));
             hb_dict_set(settings, "h", hb_value_int(orig_height));
-            hb_avfilter_append_dict(filters, "scale_qsv", settings);
+            hb_avfilter_append_dict(filters, "vpp_qsv", settings);
         }
         else
 #endif

--- a/libhb/enc_qsv.c
+++ b/libhb/enc_qsv.c
@@ -97,7 +97,6 @@ struct hb_work_private_s
 
     int                  async_depth;
     int                  max_async_depth;
-    int                  max_buffer_size;
 
     // if encode-only, system memory used
     int                  is_sys_mem;
@@ -1071,7 +1070,7 @@ int qsv_enc_init(hb_work_private_t *pv)
     }
 
     // allocate tasks
-    qsv_encode->p_buf_max_size = pv->max_buffer_size * 1000 * ( 0 == pv->param.videoParam->mfx.BRCParamMultiplier ? 1 : pv->param.videoParam->mfx.BRCParamMultiplier);
+    qsv_encode->p_buf_max_size = pv->param.videoParam->mfx.BufferSizeInKB * 1000 * ( 0 == pv->param.videoParam->mfx.BRCParamMultiplier ? 1 : pv->param.videoParam->mfx.BRCParamMultiplier);
     qsv_encode->tasks          = hb_qsv_list_init(HAVE_THREADS);
     for (i = 0; i < pv->max_async_depth; i++)
     {
@@ -1705,7 +1704,7 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
         return -1;
     }
 #endif
-
+    /* MFXVideoENCODE_Init with desired encoding parameters */
     sts = MFXVideoENCODE_Init(session, pv->param.videoParam);
 // workaround for the early 15.33.x driver, should be removed later
 #define HB_DRIVER_FIX_33
@@ -1733,6 +1732,7 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
         MFXClose(session);
         return -1;
     }
+    /* Prepare the structures for query */
     memset(&videoParam, 0, sizeof(mfxVideoParam));
     videoParam.ExtParam = extParamArray;
     videoParam.NumExtParam = 0;
@@ -1781,6 +1781,7 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
     {
         videoParam.ExtParam[videoParam.NumExtParam++] = (mfxExtBuffer*)hyper_encode;
     }
+    /* Query actual encoding parameters after MFXVideoENCODE_Init, some of them could be overridden */
     sts = MFXVideoENCODE_GetVideoParam(session, &videoParam);
     if (sts != MFX_ERR_NONE)
     {
@@ -1871,7 +1872,7 @@ int encqsvInit(hb_work_object_t *w, hb_job_t *job)
         hb_error("encqsvInit: log_encoder_params failed (%d)", err);
         return -1;
     }
-    pv->max_buffer_size = videoParam.mfx.BufferSizeInKB;
+    pv->param.videoParam->mfx = videoParam.mfx;
     // AsyncDepth has now been set and/or modified by Media SDK
     // fall back to default if zero
     pv->max_async_depth = videoParam.AsyncDepth ? videoParam.AsyncDepth : HB_QSV_ASYNC_DEPTH_DEFAULT;

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -2175,8 +2175,8 @@ int hb_qsv_full_path_is_enabled(hb_job_t *job)
         return 0;
     }
 
-    // scale_qsv filter can't convert from full to limited range
-    if (job->title->color_range == AVCOL_RANGE_JPEG && job->color_range == AVCOL_RANGE_MPEG)
+    // vpp_qsv filter can't convert from full to limited range, fallback to sw filters until unsupported
+    if (job->title->color_range == AVCOL_RANGE_JPEG)
     {
         return 0;
     }

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -1343,14 +1343,12 @@ static int query_capabilities(mfxSession session, int index, mfxVersion version,
                 info->capabilities |= HB_QSV_CAP_HYPERENCODE;
             }
         }
-        if (info->codec_id == MFX_CODEC_AV1)
+        if ((lowpower == MFX_CODINGOPTION_ON) && (info->codec_id == MFX_CODEC_AV1))
         {
             init_video_param(&videoParam);
             videoParam.mfx.CodecId = info->codec_id;
-            if (hb_qsv_implementation_is_hardware(info->implementation))
-            {
-                videoParam.mfx.LowPower = lowpower;
-            }
+            videoParam.mfx.LowPower = lowpower;
+
             init_ext_av1bitstream_option(&extAV1BitstreamParam);
             videoParam.ExtParam    = videoExtParam;
             videoParam.ExtParam[0] = (mfxExtBuffer*)&extAV1BitstreamParam;

--- a/libhb/qsv_memory.c
+++ b/libhb/qsv_memory.c
@@ -130,10 +130,36 @@ int qsv_copy_buffer_to_surface(mfxFrameSurface1* dst, hb_buffer_t* src)
 {
     uint8_t* out_luma = dst->Data.Y;
     uint8_t* out_chroma = dst->Data.VU;
+    int      out_pitch = dst->Data.Pitch;
+    int      out_height = dst->Info.Height;
 
-    // p010 and nv12
-    memcpy(out_luma, src->plane[0].data, src->plane[0].height * src->plane[0].stride);
-    memcpy(out_chroma, src->plane[1].data, src->plane[1].height * src->plane[1].stride);
+    // input buffer alignment from FFmpeg for p010 and nv12 could be different from surface alignment
+    if (out_pitch == src->plane[0].stride)
+    {
+        int height = FFMIN(out_height, src->plane[0].height);
+        memcpy(out_luma, src->plane[0].data, height * src->plane[0].stride);
+        height = FFMIN(out_height, src->plane[1].height);
+        memcpy(out_chroma, src->plane[1].data, height * src->plane[1].stride);
+    }
+    else
+    {
+        int pitch = FFMIN(out_pitch, src->plane[0].stride);
+        int height = FFMIN(out_height, src->plane[0].height);
+        for (int i = 0; i < height; i++)
+        {
+            uint8_t *out_row = out_luma + (i * out_pitch);
+            uint8_t *in_row = src->plane[0].data + i * src->plane[0].stride;
+            memcpy(out_row, in_row, pitch);
+        }
+        pitch = FFMIN(out_pitch, src->plane[1].stride);
+        height = FFMIN(out_height / 2, src->plane[1].height);
+        for (int i = 0; i < height; i++)
+        {
+            uint8_t *out_row = out_chroma + (i * out_pitch);
+            uint8_t *in_row = src->plane[1].data + i * src->plane[1].stride;
+            memcpy(out_row, in_row, pitch);
+        }
+    }
 
     return 0;
 }

